components/resources/library/src/commonMain/kotlin/org/jetbrains/compose/resources/ResourceReader.kt

```diff
@@ -0,0 +1,26 @@
+package org.jetbrains.compose.resources
+
+import androidx.compose.runtime.staticCompositionLocalOf
+
+class MissingResourceException(path: String) : Exception("Missing resource with path: $path")
+
+/**
+ * Reads the content of the file at the specified path and returns it as a byte array.
+ *
+ * @param path The path of the file to read in the resource's directory.
+ * @return The content of the file as a byte array.
+ */
+@ExperimentalResourceApi
+expect suspend fun readBytes(path: String): ByteArray
+
+internal interface ResourceReader {
```

Comment:
What do you think, if we make it public? Users can implement their own resource stores then (database, network, files, archive file)

Reply:
Ok